JK Hera/src/main/java/ayds/hera/songinfo/home/view/SongToReleaseDateStringMapper.kt

```diff
@@ -0,0 +1,49 @@
+package ayds.hera.songinfo.home.view
+
+import ayds.hera.songinfo.home.model.entities.*
+import java.time.Month
+
+interface SongToReleaseDateStringMapper {
+    fun map():String
+}
+
+internal class SongToDayReleaseDateStringMapper(private val song:Song) : SongToReleaseDateStringMapper {
+    override fun map():String {
+        val componentDate = song.releaseDate.split("-")
+        val year = componentDate[0]
+        val month = componentDate[1]
+        val day = componentDate[2]
+        return "$day/$month/$year"
+    }
+}
+
+internal class SongToMonthReleaseDateStringMapper(private val song:Song) : SongToReleaseDateStringMapper {
+    override fun map():String {
+        val componentDate = song.releaseDate.split("-")
+        val year = componentDate[0]
+        val month = componentDate[1]
+        return "${Month.of(month.toInt()).toString().toLowerCase().capitalize()}, $year"
+    }
+}
+
+internal class SongToYearReleaseDateStringMapper(private val song:Song) : SongToReleaseDateStringMapper {
+    override fun map():String {
+        val year = song.releaseDate.toInt()
+        var textToReturn:String
+
+        if(isLeapYear(year))
+            textToReturn ="(is a leap year)"
+        else
+            textToReturn ="(not a leap year)"
+
+        return "$year $textToReturn"
```

Comment:
Se puede simplificar con kotlin: 
```suggestion
        return "$year ${
        if(isLeapYear(year)) "(is a leap year)"
        else "(not a leap year)"}""
```